### DISCUSSION TITLE
Calculation of "min profit" for a crafted item

### DISCUSSION THREAD

**Discussion by Acanis87** [sansarip/nw-calculator#3] [Ideas]
Hey, its really a nice tool. Thank you^^!

I would love to see a way to add a price for every (raw) material and maybe the tax for buying the materials and for selling the crafted item.

So I get a price, what minimum price I would get, if I buy all materials to craft and sell an item. Great to decide, if its worth crafting some tools or bags in the actual market.

**Comment by sansarip** [comment]
Thanks for the input 🙌! My inspiration for [newworldcraft.io](newworldcraft.io) was this 👉 [Guild Wars 2 crafting calculator](https://gw2efficiency.com/crafting/calculator) which does include trading information, as you're suggesting! So, it has been on my radar 💡  

The reason why I haven't implemented this already is because Amazon hasn't released a public API for New World (yet), and it may not come out anytime soon 😢 I'd imagine it's pretty low on their priority list atm. As far as I'm aware, data like trading info and the like is dynamic and can't be datamined like the rest of the data that I scrape from [nwdb.info](nwdb.info) 😞 

That being said, while it's impractical to get trading data right now, it's not impossible. The game client does communicate with Amazon's servers to get info like trading data, and a tool like [wireshark](https://www.wireshark.org/) could help me pinpoint what network requests the game client makes to get that sweet-sweet data 🔍 

In summary, I'll look into it, but it's probably highly unlikely I'll be able to complete this feature until Amazon releases a public API for the game 🚀

**Comment by sammullican** [comment]
I was about to submit a similar discussion topic for this, and as an alternative suggestion in a similar vein, how about potentially doing something like allowing the user to manually input prices for materials based on their server and trading posts as they're looking to allow for price comparisons.

This obviously would not be the total price due to taxes, but it would give the user a good idea regarding what their total investment will be close to.

For example, if you look for 1 Steel Ingot, it is structured as follows:

1 Steel Ingot
- 1 Sand Flux
- 3 Iron Ingot
    - 12 Iron Ore
- 2 Charcoal
    - 4 Green Wood

Having an input box next to each of these materials or a separate module that lists:
- Steel Ingot Price Per Unit: __
- Sand Flux Price Per Unit: __
- Iron Ingot Price Per Unit: __
- Iron Ore Price Per Unit: __
- Charcoal Price Per Unit: __
- Green Wood Price Per Unit: __

The entered value would be multiplied by the quantities required in the ingredient list and return something like the following in a module similarly to the ingredients summary module:
- Steel Ingot Total Cost: __
- Sand Flux Total Cost: __
- Iron Ingot Total Cost: __
- Iron Ore Total Cost: __
- Charcoal Total Cost: __
- Green Wood Total Cost: __

Buy vs. Craft
- Cost to Buy: __
- Cost to Craft: __

A step beyond this would include the cheapest combination of materials.

For example, if the Iron Ingot cost is cheaper than the Iron Ore cost, then in a "Crafting Cost Summary", "Iron Ingot" would be returned as the most cost efficient material. Whereas, if Green Wood is cheaper than Charcoal, then Green Wood would be shown in the same summary. This could look like:

Crafting Cost Summary:
- Sand Flux Cost: __
- Iron Ingot Cost: __
- Green Wood Cost: __
- Total Cost: __
- _This is the cheapest combination of materials based on user input values._

Just a couple thoughts. Thanks again for putting together this calculator, I use it daily and very much appreciated the effort you've put in on such an amazing and nice looking tool. Keep up the good work!

**Reply by sansarip** [reply]
Interesting idea! Yea, I think I could implement a feature like this 🤔 💭
